internal/rpc/rpcserver/server.go

```diff
@@ -3469,3 +3471,74 @@ func (s *walletServer) LockAccount(ctx context.Context, req *pb.LockAccountReque
 	}
 	return &pb.LockAccountResponse{}, nil
 }
+// GetPeerInfoResult models the data returned from the getpeerinfo command.
+type GetPeerInfoResult struct {
+	ID             int32  `json:"id"`
+	Addr           string `json:"addr"`
+	AddrLocal      string `json:"addrlocal"`
+	Services       string `json:"services"`
+	Version        uint32 `json:"version"`
+	SubVer         string `json:"subver"`
+	StartingHeight int64  `json:"startingheight"`
+	BanScore       int32  `json:"banscore"`
+}
+
+// getPeerInfo responds to the getpeerinfo request.
+// It gets the network backend and views the data on remote peers when in spv mode
+func (s *walletServer) GetPeerInfo(ctx context.Context, req *pb.GetPeerInfoRequest) (*pb.GetPeerInfoResponse, error) {
+	n, err := s.wallet.NetworkBackend()
+	if err != nil {
+		return nil, err
+	}
+	syncer, ok := n.(*spv.Syncer)
+	if !ok {
+		var resp []*dcrdtypes.GetPeerInfoResult
```

Comment:
you can declare this like

```
var resp []*struct {
    ID int32 `json:"id"`
    Addr string `json:"addr"`
    // ...
}
```

to avoid creating a new type and adding it to the public api.